stdlib/streams/src/main/ballerina/streams/windows.bal

```
@@ -1539,3 +1539,296 @@ public function delayWindow(any[] windowParameters, function (StreamEvent[])? ne
     DelayWindow delayWindow1 = new(nextProcessPointer, windowParameters);
     return delayWindow1;
 }
+
+public type SortWindow object {
+
+    public int lengthToKeep;
+    public any [] windowParameters;
+    public LinkedList sortedWindow;
+    public string[] sortMetadata;
+    public string[] fields;
+    public string[] sortTypes;
+    public function (StreamEvent[])? nextProcessPointer;
+
+    public function __init(function (StreamEvent[])? nextProcessPointer, any [] windowParameters) {
+        self.nextProcessPointer = nextProcessPointer;
+        self.windowParameters = windowParameters;
+        self.sortedWindow = new;
+        self.lengthToKeep = 0;
+        self.sortMetadata = [];
+        self.fields = [];
+        self.sortTypes = [];
+        self.initParameters(windowParameters);
+    }
+
+    public function initParameters(any[] parameters) {
+        if(parameters.length() >= 3 && parameters.length() % 2 == 1) {
```

Can u invert the 'if' condition and follow fail fast approach. ?
```suggestion
        if(!(parameters.length() >= 3 && parameters.length() % 2 == 1)) {
        error err = error("Sort window should have two or more " +
                 "parameters (<int> windowLength, <string> attribute1, <string> order1, " +
                 "<string> attribute2, <string> order2, ...), but found " + parameters.length()
                 + " input attributes" );
             panic err;
             }
```